File: examples/python/02_install_skills.py
"""Example: Install skills from the demo-starter bundle into a Claude Code project."""

from pathlib import Path

import musher

# Pull the demo-starter bundle
bundle = musher.pull("musher-dev/demo-starter:1.0.0")

# Target directory for Claude Code skills
# Relative to your project root — Claude Code discovers skills from .claude/skills/
skills_dir = Path(".claude/skills")

# Install specific skills from the bundle.
# clean=True removes stale Musher-managed installs for this bundle from the
# target directory; it does NOT affect manually placed skills.
bundle.install_claude_skills(
    skills_dir,
    skills=["summarize-changes", "write-commit-message"],
    clean=True,
)

print(f"Installed skills to {skills_dir}/")
for skill in bundle.skills():
    installed_path = skills_dir / f"{skill.name}.md"
    status = "✓" if installed_path.exists() else "✗"
    print(f"  {status} {skill.name}")

print()
print("Skills are now available in Claude Code.")
print("Try asking: 'Summarize the changes since the last release'")
print("Or: 'Write a commit message for my staged changes'")

File: examples/typescript/02-install-skills.ts
/**
 * Install skills from the repo-documentation-governance bundle into a Claude Code project.
 *
 * This copies multi-file skills (SKILL.md + companion files like rubrics,
 * templates, and checklists) into .claude/skills/ where the Claude Agent SDK
 * discovers them.
 *
 * Prerequisites:
 *   export MUSHER_API_KEY="mush_..."
 *
 * To use your own published bundle, set:
 *   export MUSHER_BUNDLE_REF="your-namespace/repo-documentation-governance:1.0.0"
 *
 * Run:
 *   npx tsx 02-install-skills.ts
 */

import { pull } from "@musher-dev/musher-sdk";

const bundleRef =
	process.env.MUSHER_BUNDLE_REF ?? "musher-examples/repo-documentation-governance:1.0.0";

const bundle = await pull(bundleRef);

// installClaudeSkills writes to .claude/skills/ under the given project root.
const projectRoot = ".";

const written = await bundle.installClaudeSkills(projectRoot);

console.log(`Installed ${written.length} file(s) to .claude/skills/`);
for (const filePath of written) {
	console.log(`  ${filePath}`);
}

console.log();
console.log("Skills are now available to the Claude Agent SDK.");
console.log("See examples/python/repo_docs_audit.py for the Agent SDK demo.");
